src/registry/LocalRegistry.ts

```diff
@@ -99,4 +160,37 @@ export class LocalRegistry extends BaseRegistry implements IRegistry {
 
     return filePaths.flat();
   }
+
+  protected createFile(file: { filePath: string; data: string }): void {
+    const dirPath = path.dirname(file.filePath);
+    if (!fs.existsSync(dirPath))
+      fs.mkdirSync(dirPath, {
+        recursive: true,
+      });
+    fs.writeFileSync(file.filePath, file.data);
+  }
+
+  protected createChainFile(
+    chainName: ChainName,
+    fileName: keyof ChainFiles,
+    data: any,
+    cache: ChainMap<any>,
+    prefix?: string,
+  ) {
+    const filePath = path.join(this.uri, this.getChainsPath(), chainName, `${fileName}.yaml`);
+    this.listContentCache!.chains[chainName] ||= {};
+    this.listContentCache!.chains[chainName][fileName] = filePath;
+    cache[chainName] = data;
+    this.createFile({ filePath, data: toYamlString(data, prefix) });
+  }
+
+  protected removeFiles(filePaths: string[]): void {
+    for (const filePath of filePaths) {
+      fs.unlinkSync(filePath);
```

Comment:
is this a lock?

Reply:
No, this is how you delete a file with node's fs lib. Strange I know.